### PR TITLE
Added auditing for Practitioners and Practices.

### PR DESCRIPTION
**Pull request by apigram**
Removed unnecessary override of layout property in GpController and PracticeController.
Added missing redirect when saving a Practice without an address.

### CODE REVIEW

File: protected/controllers/GpController.php [hunk 1]
```diff
@@ -86,6 +84,7 @@ public function actionCreate($context = null)
 
     public function performGpSave(Contact $contact, Gp $gp, $isAjax = false)
     {
+        $action = $gp->isNewRecord ? 'add' : 'edit';
```
Comment (marshl) [inline protected/controllers/GpController.php]: should be getIsNewRecord() I think, isNewRecord is the internal value

File: protected/controllers/GpController.php [hunk 2]
```diff
@@ -104,26 +103,27 @@ public function performGpSave(Contact $contact, Gp $gp, $isAjax = false)
...
                 if ($gp->save()) {
                     $transaction->commit();
+                    Audit::add('Gp', $action . '-gp', "Practitioner manually [id: $gp->id] {$action}ed.");
                     if (!$isAjax) {
                         $this->redirect(array('view', 'id' => $gp->id));
                     }
                 } else {
                     if ($isAjax) {
-                        throw new CHttpException(400,"Unable to save Gp contact");
+                        throw new CHttpException(400,"Unable to save Practitioner contact");
```
Comment (marshl) [inline protected/controllers/GpController.php]: So this is feature/cera only?

File: protected/controllers/PracticeController.php [hunk 3]
```diff
@@ -111,36 +108,41 @@ public function performPracticeSave(Contact $contact, Practice $practice, Addres
...
+                    throw new CHttpException(400, "Unable to save Practice contact");
                 }
                 $transaction->rollback();
             }
         } catch (Exception $ex) {
             OELog::logException($ex);
             $transaction->rollback();
             if ($isAjax) {
-                throw new CHttpException(400,"Unable to save Practice contact");
+                throw new CHttpException(400, "Unable to save Practice contact");
```
Comment (marshl) [inline protected/controllers/PracticeController.php]: Double quotes are unnecessary here

File: protected/controllers/PracticeController.php [hunk 4]
```diff
@@ -198,7 +200,7 @@ public function actionIndex()
      * Returns the data model based on the primary key given in the GET variable.
      * If the data model is not found, an HTTP exception will be raised.
      * @param integer $id the ID of the model to be loaded
-     * @return Practice the loaded model
+     * @return Practice|CActiveRecord the loaded model
```
Comment (marshl) [inline protected/controllers/PracticeController.php]: Huh? When would it ever return the CActiveRecord base class?

**Review by marshl — APPROVED** [review]

File: protected/controllers/GpController.php (hunk 1 above)
Comment (apigram) [inline protected/controllers/GpController.php]: Really? I've seen this pattern used elsewhere.

File: protected/controllers/GpController.php (hunk 2 above)
Comment (apigram) [inline protected/controllers/GpController.php]: Yeah. For the moment.

File: protected/controllers/PracticeController.php (hunk 3 above)
Comment (apigram) [inline protected/controllers/PracticeController.php]: I'm surprised PSR-2 auto-correction didn't replace those.

File: protected/controllers/PracticeController.php (hunk 4 above)
Comment (apigram) [inline protected/controllers/PracticeController.php]: For some reason findByPk returns CActiveRecord, not the subclass.

**Pull request merged**